src/ClientTest.hs

```diff
@@ -0,0 +1,34 @@
+import Control.Exception (tryJust)
+import Control.Monad (guard, void)
+import System.Process
+import System.Directory (doesFileExist, removeFile)
+import System.Exit
+import System.IO.Error (isDoesNotExistError)
+
+import Network (sClose, Socket)
+
+import Server
+
+clientBin :: FilePath
+clientBin = "./dist/build/hdevtools/hdevtools"
```

Comment:
I support your idea of tests, although this won't work for half of our travis jobs.